daemon/containerd/image_list.go

```diff
@@ -218,6 +213,14 @@ func (i *ImageService) Images(ctx context.Context, opts imagetypes.ListOptions)
 			totalSize += ts
 			allChainsIDs = append(allChainsIDs, chainIDs...)
 
+			if opts.ContainerCount {
+				i.containers.ApplyAll(func(c *container.Container) {
+					if c.ImageManifest != nil && c.ImageManifest.Digest == img.Target().Digest {
```

Comment:
For a follow-up; we should really document the `ImageManifest` field in container; I went looking to see if this was the "platform-specific" manifest, or if this was the manifest-index ("root" of the image), but the field is currently not documented.